cpp/open3d/t/geometry/PointCloud.cpp

```diff
@@ -259,8 +259,13 @@ open3d::geometry::PointCloud PointCloud::ToLegacyPointCloud() const {
                 core::eigen_converter::TensorToEigenVector3dVector(GetPoints());
     }
     if (HasPointColors()) {
-        pcd_legacy.colors_ = core::eigen_converter::TensorToEigenVector3dVector(
-                GetPointColors());
+        auto colors = GetPointColors();
```

Comment:
We may use `core::Tensor` instead of `auto` here.

Reply:
`colors = GetPointColors()` the underlying memory blob is shared. 
`colors = colors.To(core::Dtype::Float64) / 255.0` <- might modify the memory blob associated with the pointcloud. 
We may use `colors = GetPointColors().Clone()` for safety and simplicity. 

If we want to avoid copy in case we don't need to convert. We may do this:
```
core::Tensor colors;
if(colors.GetDtype() == core::Dtype::UInt8) {
    colors = GetPointColors().To(core::Dtype::Float64) / 255.0;
} else if ( ) {

} else {
    colors = GetPointColors();
}
pcd_legacy.colors_ =
                core::eigen_converter::TensorToEigenVector3dVector(colors);
```

Reply:
Yes, the alternative to `Clone()` is `To(core::Dtype::Float64, /*copy=*/true)`, this will also copy the original tensor contents.